synapse/federation/transport/server.py

```diff
@@ -101,19 +103,23 @@ def authenticate_request(self, request, content):
 
         def parse_auth_header(header_str):
             try:
-                params = auth.split(" ")[1].split(",")
-                param_dict = dict(kv.split("=") for kv in params)
+                params = auth.split(b" ")[1].split(b",")
+                param_dict = dict(kv.split(b"=") for kv in params)
 
                 def strip_quotes(value):
-                    if value.startswith("\""):
+                    if value.startswith(b"\""):
                         return value[1:-1]
                     else:
                         return value
 
-                origin = strip_quotes(param_dict["origin"])
-                key = strip_quotes(param_dict["key"])
-                sig = strip_quotes(param_dict["sig"])
-                return (origin, key, sig)
+                origin = strip_quotes(param_dict[b"origin"])
+                key = strip_quotes(param_dict[b"key"])
+                sig = strip_quotes(param_dict[b"sig"])
+                if PY3:
```

Comment:
I'm curious as to why this needs to be conditional.

Reply:
It isn't (I was chasing another bug, this obviously wasn't what fixed it)